templates/template.yaml

```diff
@@ -0,0 +1,22 @@
+outdir: "output/"
+tsv : "./templates/template.csv"
```

Comment:
what would happen if different h5ad files in the .csv have different h5ad structure/embedding naming rule etc. ?

Reply:
There are two steps where that could cause issues,
1.  When processing the `h5ad` file, if `compute_embeddings` is set to `True` the script will look for the default embedding names from scanpy, so `"X_umap"` and `"X_pca"` in `obsm` and compute them if they do not exist. As they're scanpy output the script won't modify the names and there's no option for it. In case the embeddings have already been computed with different names or stored in a different structure then `compute_embeddings` should be set to `False`.
2. The [vitessce options](https://github.com/haniffalab/vitessce-pipeline/blob/df2d21a24fffe7bdb394b37c3e07552f9db22bdd/templates/template.yaml#L10) should be manually set by the user who knows how data is structured. If set incorrectly Vitessce won't be able to load the view config.

If there is a mix of `h5ad`'s  with "conflicting" structures/names then both the `args` for the `h5ad` processing function and the vitessce options can be set independently. The `h5ad` `args` can be set for each _file_ in the `csv` easily; the vitessce options can be set for each _dataset_ in the `csv` (this can be a bit messy as it would be a pretty long json string).
So the `csv` _can_ accommodate multiple `h5ad`'s with different structures, but if the differences require setting dataset-specific vitessce options the better option would be writing another `yaml` file with those settings.

Reply:
thanks Daniela! I see, that makes sense. Closing this and will merge